### PR TITLE
[Community][fix] Fix Azure cosmos db no SQL similarity search with score and mmr 

### PR DESCRIPTION
**Pull request by wassim-mechergui-shift**
**Issues**:
* similarity search with score not working for `VectorStoreRetriever` instanciated from `AzureCosmosDBNoSqlVectorSearch` because it requires an implementation of  `__select_relevance_score_fn_`. 
* mmr retrieval is not working because the fetching of metadata field from item to document metadata is currently bugged : (issues: https://github.com/langchain-ai/langchain/issues/26097, https://github.com/langchain-ai/langchain/issues/28476)

**Changes**:
* VectorStoreRetriever fix:
  *  For the similarity search from `VectorStoreRetriever` a rather typical implementation was added ([example for Qdrant](https://github.com/langchain-ai/langchain/blob/ff675c11f6814e88aa338dc3668aa84d1ade3d19/libs/community/langchain_community/vectorstores/qdrant.py#L1928))
* with_embeddings to similarity search:
  * Fix for the with_embedding keyword) due to a change made here (https://github.com/langchain-ai/langchain/issues/26097), item no longer has `self._embedding_key` and has "$1" (that represents it's order in the parameters). Since Cosmos DB’s SQL API does not support parameterized aliases or dynamic aliasing within the query, we chose to name it 'embeddingKey'
  * optimization : if with_embedding is not True, we don't need to query the embeddings. Querying the embeddings is heavy and makes requests much slower. This was changed
* Changed initialisation of vector policy (from which we get the embedding key and the distance function)
  * if user specifies a dict, we check it has the correct form
  * if user specifies the container creation if it doesn't exist option, we check this is not null
  * for the final user vector policy property
    *  if the container exposes one, we use that one  (and raise a warning if the user specified policy has differences)
    * if the container doesn't expose one, we use the user exposed one (with an error in case nothing was specified)

These changes were tested locally.

### CODE REVIEW

File: libs/community/langchain_community/vectorstores/azure_cosmos_db_no_sql.py [hunk 1]
```diff
@@ -121,6 +121,9 @@ def __init__(
         self._embedding_key = self._vector_embedding_policy["vectorEmbeddings"][0][
             "path"
         ][1:]
+        self._distance_strategy = self._vector_embedding_policy["vectorEmbeddings"][0][
+            "distanceFunction"
```
Comment (baskaryan) [inline libs/community/langchain_community/vectorstores/azure_cosmos_db_no_sql.py:125]: is this guaranteed to always be specified? seems like we only check that self._vector_embedding_policy isn't empty if self._create_container is True

Comment (wassim-mechergui-shift) [reply libs/community/langchain_community/vectorstores/azure_cosmos_db_no_sql.py:146]: Actually, this assumption is not always valid 
> Once created, it's possible to query for these info in the properties of the container. (through container.read())

It seems there is a bug in azure cosmos db no sql but if the container was the result of a partition_key migration then it seems the information on the vector policy is lost (so not possible to read through `container.read()`). However, the container still fully supports vector search !

I'll readapt the implementation to account for this case.

I'm sorry for the long post I wanted to be thorooughly descriptive and argumentative in my implementation choices!

**Review by ccurme — COMMENTED** [review]
@wassim-mechergui-shift it looks like there's some redundancy with https://github.com/langchain-ai/langchain/pull/28716.

Closing for now but happy to re-open if there are differences in functionality and you'd like to merge with latest changes (ok to open another PR as well).

**Comment by wassim-mechergui-shift** [comment]
Hi @ccurme ! Thanks for the comment. I didn't notice the other PR.

I will test when I get time from my side if the bugs I mentionned here were addressed in that PR (and hopefully no additional bugs were added). If so I'll merge here with the latest changes

**Pull request closed without merging**